Feature/in a table

emclaug2 opened the pull request:
<!-- Include a bulleted list summarizing the main changes you have made in this PR -->
#### Changes proposed in this Pull Request:
- Form validation in a table example 

- There's currently some weird behavior with the table; I've never used a MUI Table before, if you hover over the 2nd row there's some hover-state styles that appear.  These styles are only applied to the odd-numbered rows & may be an issue with the theme.

jeffvg commented:
looks good but yeah has that one weird hover state. Looks like there is a prop for it..
https://material-ui.com/api/table-row/

jeffvg commented:
looks like that second row inherits different background color. Might want to use data elements other than power xpert blue

ektasawant commented:
Looks good 👍 .

Few comments from my side: 
1. Note is missing below table `Remember that in a real application ...`
2. Table width is not matching with design.
3. In mobile color of `#1, #2 and #3` is gray-500
4. Can we remove divider prop from last column?

huayunh commented:
![image](https://user-images.githubusercontent.com/8997218/130611778-ea6872b2-c9c5-4b80-a07d-a3bbd216cd5d.png)
Number cells shall be right-aligned. Text cells shall stay left-aligned



![image](https://user-images.githubusercontent.com/8997218/130611970-1b6d2497-23c9-4a3f-ba65-6d303743a5fb.png)
What is this white space on the bottom for?

emclaug2 commented:
> looks like that second row inherits different background color. Might want to use data elements other than power xpert blue

These style issues were actually being inherited from the "Responsive Table" List example.  I've fixed the style bleed so we don't get any unwanted hover effects now.

emclaug2 commented:
> Looks good 👍 .
> 
> Few comments from my side:
> 
> 1. Note is missing below table `Remember that in a real application ...`
> 2. Table width is not matching with design.
> 3. In mobile color of `#1, #2 and #3` is gray-500
> 4. Can we remove divider prop from last column?


I've added the missing text, adjusted the width, updated the index color of list view, and made the last divider in the list view full-width.

emclaug2 commented:
> Number cells shall be right-aligned. Text cells shall stay left-aligned
You got it, should be good now. 

> What is this white space on the bottom for?
It's to keep you on your toes.  It's been removed :)

jeffvg reviewed (approved):
looks good

huayunh commented on src/pages/form-validation/table/index.tsx:
```diff
@@ -0,0 +1,222 @@
+
+const useStyles = makeStyles((theme: Theme) => ({
+    toolbarGutters: {
+        padding: '0 16px',
```
```suggestion
        padding: `0 ${theme.spacing(2)}px`,
```

huayunh commented on src/pages/form-validation/table/index.tsx:
```diff
@@ -0,0 +1,222 @@
+    const getTable = (): JSX.Element => (
+        <TableContainer component={Paper} className={classes.tableContainer}>
+            <Table>
+                <TableHead style={{ fontWeight: 600 }}>
```
may want to log an enhancement and make this into theme

emclaug2 replied:
Good callout, this style actually was already provided as the default.  No issue needed & this inline style was removed.

huayunh commented on src/pages/form-validation/table/index.tsx:
```diff
@@ -0,0 +1,222 @@
+                    <div style={{ display: 'flex', alignItems: 'center', height: 64 }}>
+                        <Typography
+                            variant={'body1'}
+                            style={{ width: '20%', paddingLeft: theme.spacing(2), color: Colors.gray[500] }}
```
in the future we might make this dark theme. Can we do `theme.palette.text.secondary`? Avoid citing `@pxblue/colors` directly.

emclaug2 replied:
The @pxblue/colors import has been removed for this example.

huayunh commented on src/pages/form-validation/table/table-validation.test.tsx:
```diff
@@ -0,0 +1,24 @@
+/* eslint-disable */
```
why eslint-disable here?

emclaug2 replied:
I think this was a copy/paste gone wrong.  This file was already linted corrected & the comment was removed.

huayunh commented:
## Alignment
"ID" shall be right-aligned, and "Name" shall be left-aligned
![image](https://user-images.githubusercontent.com/8997218/131412468-432d71cb-fbf0-42b9-8cf7-a4d37b82b372.png)

## Mobile List Styling
As for the mobile list view styling, I suggest this:
![image](https://user-images.githubusercontent.com/8997218/131413554-792c8b63-9299-4c25-a7f9-e0db148fa7fc.png)

![image](https://user-images.githubusercontent.com/8997218/131413917-207d8ac4-dd2d-4fe7-805a-20a1e2bc8902.png)

The idea here is that these elements are lined up
![image](https://user-images.githubusercontent.com/8997218/131414096-92b55e10-02f2-442a-a1b2-6fce5052fdee.png)

I would avoid relative width as much as possible — your spacing shall look correct at a minimum width of 320px.

## Inline Styles
I noticed that you have a lot of inline styles. I would suggest pull them out and put them together in the createMuiTheme function. If you do it this way, your code snippet gets less crowded, making it easier to sort out the code structure. Use inline styles only when you have substantial trouble with the `className` approach.

emclaug2 commented:
This demo has been refactored to use InfoListItems and mobile view should match angular now.

ektasawant commented:
Changes looks good .

huayunh commented on src/pages/form-validation/table/index.tsx:
```diff
@@ -0,0 +1,198 @@
+                            />
+                        }
+                    />
+                    <InfoListItem
+                        title={''}
+                        style={{ height: theme.spacing(2), marginBottom: -theme.spacing(2) }}
+                        divider={index === rows.length - 1 ? 'full' : 'partial'}
+                    />
```
Feels hacky.... you introduced a negative margin + a blank ILI to do the divider. The end result looks visually good though. Wonder how @joebochill feels about this.

emclaug2 replied:
Ok so the negative margin wasn't required; but the 16px InfoListItem is still there just so we don't have to do any calculations to figure out how big a partial divider should be.

huayunh reviewed (commented):
> ## Alignment
>"ID" shall be right-aligned, and "Name" shall be left-aligned
> ![image](https://user-images.githubusercontent.com/8997218/131412468-432d71cb-fbf0-42b9-8cf7-a4d37b82b372.png)

joebochill commented on src/pages/form-validation/table/index.tsx:
```diff
@@ -0,0 +1,198 @@
+                            />
+                        }
+                    />
+                    <InfoListItem
+                        title={''}
+                        style={{ height: theme.spacing(2) }}
+                        divider={index === rows.length - 1 ? 'full' : 'partial'}
+                    />
```
```suggestion
                    <Divider variant={'inset'} style={{marginTop: theme.spacing(1), marginBottom: theme.spacing(1)}} />
```
